src/together/resources/endpoints.py

```diff
@@ -0,0 +1,314 @@
+                disable_prompt_cache=disable_prompt_cache,
+                disable_speculative_decoding=disable_speculative_decoding,
+                state=state,
+            )
+            return await self._api.create_endpoint(create_endpoint_request=request)
+
+        return self._loop.run_until_complete(_create())
+
+    def list(
+        self, type: Literal["dedicated", "serverless"] | None = None
+    ) -> List[ListEndpoint]:
+        """
+        List all endpoints.
+
+        Args:
+            type (str, optional): Filter endpoints by type ("dedicated" or "serverless"). Defaults to None.
+
+        Returns:
+            Dict[str, Any]: Response containing list of endpoints in the data field
+        """
+
+        async def _list() -> List[ListEndpoint]:
+            response = await self._api.list_endpoints(type=type)
+            return response.data
+
+        return self._loop.run_until_complete(_list())
+
+    def get(self, endpoint_id: str) -> DedicatedEndpoint:
+        """
+        Get details of a specific endpoint.
+
+        Args:
+            endpoint_id (str): ID of the endpoint to retrieve
+
+        Returns:
+            DedicatedEndpoint: Object containing endpoint information
+        """
+
+        async def _get() -> DedicatedEndpoint:
+            return await self._api.get_endpoint(endpoint_id=endpoint_id)
+
+        return self._loop.run_until_complete(_get())
+
+    def delete(self, endpoint_id: str) -> None:
+        """
+        Delete a specific endpoint.
+
+        Args:
+            endpoint_id (str): ID of the endpoint to delete
+        """
+
+        async def _delete() -> None:
+            return await self._api.delete_endpoint(endpoint_id=endpoint_id)
+
+        return self._loop.run_until_complete(_delete())
+
+    def update(
+        self,
+        endpoint_id: str,
+        *,
+        min_replicas: Optional[int] = None,
+        max_replicas: Optional[int] = None,
+        state: Optional[Literal["STARTED", "STOPPED"]] = None,
+        display_name: Optional[str] = None,
+    ) -> DedicatedEndpoint:
+        """
+        Update an endpoint's configuration.
+
+        Args:
+            endpoint_id (str): ID of the endpoint to update
+            min_replicas (int, optional): The minimum number of replicas to maintain
+            max_replicas (int, optional): The maximum number of replicas to scale up to
+            state (str, optional): The desired state of the endpoint ("STARTED" or "STOPPED")
+            display_name (str, optional): A human-readable name for the endpoint
+
+        Returns:
+            DedicatedEndpoint: Object containing endpoint information
+        """
+
+        async def _update() -> DedicatedEndpoint:
+            kwargs: Dict[str, Any] = {}
+            if min_replicas is not None or max_replicas is not None:
+                current_min = min_replicas
+                current_max = max_replicas
+                if current_min is None or current_max is None:
```

Comment:
This check is not needed as the api supports updating just one of min or max replicas. There is no harm in doing a check here but then we can avoid the get call.